src/parsers/intersection.ts

```diff
@@ -21,5 +28,36 @@ export function parseIntersectionDef(
     }),
   ].filter((x): x is JsonSchema7Type => !!x);
 
-  return allOf.length ? { allOf } : undefined;
+
+  let unevaluatedProperties: { unevaluatedProperties: boolean } | undefined = {unevaluatedProperties: false};
+  const mergedAllOf: JsonSchema7Type[] = []
+  // If either of the schemas is an allOf, merge them into a single allOf
+  allOf.forEach((schema) => {
+    if (isJsonSchema7AllOfType(schema)) {
+
+      mergedAllOf.push(...schema.allOf);
+      if (schema.unevaluatedProperties === undefined) {
+        // If one of the schemas has no unevaluatedProperties set,
+        // the merged schema should also have no unevaluatedProperties set
+        unevaluatedProperties = undefined;
+      }
+
+    } else {
+
+      let nestedSchema: JsonSchema7Type = schema
+      if ('additionalProperties' in schema && schema.additionalProperties === false) {
+        const {additionalProperties, ...rest} = schema;
+        nestedSchema = rest;
+      } else {
+        // As soon as one of the schemas has additionalProperties set not to false, we allow unevaluatedProperties
+        unevaluatedProperties = undefined;
+      }
+      mergedAllOf.push(nestedSchema);
+
+    }
+  });
+  return mergedAllOf.length ? {
+    allOf: mergedAllOf,
+    ...unevaluatedProperties
```

Comment:
@StefanTerdell the `unevaluatedProperties` property was introduced in JsonSchema `2019-09`, but this library is generating `Draft-7`. Should we upgrade to `2019-09` or should I remove this new field?